src/app/game/services/example-drawing.service.ts

```diff
@@ -2,17 +2,32 @@ import { ExampleDrawingsData } from '@/app/shared/models/backend-interfaces';
 import { endpoints } from '@/app/shared/models/endpoints';
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { Observable } from 'rxjs';
+import { SupportedLanguages } from '@/app/shared/models/interfaces';
 
 @Injectable({
   providedIn: 'root',
 })
 export class ExampleDrawingService {
-  baseUrl = endpoints.TEKNISKBACKEND;
+  private readonly baseUrl = endpoints.TEKNISKBACKEND;
+  private exampleDrawings: string[] = [];
 
   constructor(private http: HttpClient) {}
 
-  getExampleDrawings(body: ExampleDrawingsData): Observable<string[]> {
-    return this.http.post<string[]>(`${this.baseUrl}/${endpoints.GETEXAMPLEDRAWINGS}`, body);
+  preLoadExampleDrawings(numberOfImages: number, label: string, lang: SupportedLanguages) {
+    const body: ExampleDrawingsData = {
+      game_id: '',
+      number_of_images: numberOfImages,
+      label: label,
+      lang: lang,
+    };
+
+    this.http.post<string[]>(`${this.baseUrl}/${endpoints.GETEXAMPLEDRAWINGS}`, body).subscribe((res) => {

```

Comment:
Does this ever get unsubscribed?

Reply:
Think also best practice is to subscribe in the components, not in the service. According to what Tallang said in the Angular workshop. But I see that this has been violated in several files now

Reply:
I am glad you posted these comments, because I did not know the answer to them when I came up with this solution.

#### This is what I found out
Any http call using the http-client that angular provides automatically unsubscribes once the response is received, so we don't have to explicitly unsubscribe using the onDestroy-option in the component that uses the observable.

To answer the second comment, I completely agree that we should stick to that convention. I wanted to stick to that convention here, but I found it difficult. If the component gets destroyed _before_  the http request or socket transfer is fully completed, we may stumble upon unwanted errors, because that will just shut off the connection between the backend and frontend without considering the state of data transfer.

Also, `pipe(take(1), ... )` that is used in the multiplayerservice-method "preLoadExampleDrawings() will unsubscribe automatically, because of the "take()" pipe. It takes the data it needs, and then shuts the connection (unsibscribes).

If you know of any other files that isn't related to this PR that violates the unsubscribe convention, please let me know and I will fix 👊🏻

Also please let me know if some of the stuff I wrote was unclear!

Reply:
Very nice. Thank you for the clarification